xdsl/parser.py

```diff
@@ -1293,6 +1180,11 @@ def _parse_builtin_dense_attr(self, _name: Span) -> Attribute | None:
         self.parse_characters(":", err_msg)
         type = self.expect(self.try_parse_type,
                            "Dense attribute must be typed!")
+
+        # Make sure type is AnyTensorType
+        assert isinstance(type, VectorType | TensorType | UnrankedTensorType)
```

Comment:
Why not AnyTensorType? Can you not do isinstance on TypeVars?

I guess this is just one of the downsides when you throw a type system at a language like python.

Reply:
I would prefer isa, feels like more reuse to me. Not a strong opinion though...

Reply:
- [ ] use isa